File: frameworks/rust/axum/Implementation/routes/compressed.rs
use axum::routing::get;
use axum::{Json, Router};
use tower_http::CompressionLevel;
use tower_http::compression::CompressionLayer;

use crate::Payloads;
use crate::serial;

/// compressed: these routes answer like any other. tower-http's compression layer, on this
/// family's routes alone, gzips the answer when the request asks for it.
pub fn router(p: &'static Payloads) -> Router {
    Router::new()
        .route("/compressed/small", get(move || async move { (serial::fresh(), Json(&p.small)) }))
        .route("/compressed/large", get(move || async move { (serial::fresh(), Json(&p.large)) }))
        // rb:wiring compressed.*
        // gzip at the fastest level every framework here compresses at, and the default
        // predicate, which leaves a body under 32 bytes alone.
        .layer(CompressionLayer::new().quality(CompressionLevel::Fastest))
}

File: frameworks/python/litestar/Implementation/routes/compressed.py
from litestar import Response, Router, get
# rb:wiring compressed.*
from litestar.config.compression import CompressionConfig
from litestar.middleware import DefineMiddleware
from litestar.middleware.compression import CompressionMiddleware

from payloads import Payload, Payloads
from serial import fresh


def router(p: Payloads) -> Router:
    """compressed: Litestar's compression middleware on this family's Router alone, which gzips an
    answer when the request asks. The application's own compression_config would put it in front of
    every route."""
    # rb:wiring compressed.*
    # A body under 500 bytes goes out as it is, which is the default minimum_size, and gzip runs at
    # level 1, the fastest level every framework here compresses at. Litestar's own default is 9.
    gzip = DefineMiddleware(CompressionMiddleware, config=CompressionConfig(backend="gzip", gzip_compress_level=1))

    @get("/compressed/small")
    async def small() -> Response[Payload]:
        return Response(p.small, headers=fresh())

    @get("/compressed/large")
    async def large() -> Response[Payload]:
        return Response(p.large, headers=fresh())

    return Router(path="/", route_handlers=[small, large], middleware=[gzip])
